compaction/compaction_manager.cc

```diff
@@ -873,14 +873,16 @@ inline bool compaction_manager::task::can_proceed(throw_if_stopping do_throw_if_
     return _cm.can_proceed(_compacting_table);
 }
 
-future<stop_iteration> compaction_manager::task::maybe_retry(std::exception_ptr err) {
+future<stop_iteration> compaction_manager::task::maybe_retry(std::exception_ptr err, bool throw_on_abort) {
     try {
         std::rethrow_exception(err);
     } catch (sstables::compaction_stopped_exception& e) {
         cmlog.info("{}: {}: stopping", *this, e.what());
     } catch (sstables::compaction_aborted_exception& e) {
         cmlog.error("{}: {}: stopping", *this, e.what());
         _cm._stats.errors++;
+        if (throw_on_abort)
+            throw;
```

Comment:
coding style nit: we always use curly braces to prevent accidental bugs due to misleading indentation.
```c++
        if (throw_on_abort) {
            throw;
        }
```